gcloud/datastore/test___init__.py

```diff
@@ -141,8 +147,10 @@ def test_set_gcd_and_production_env_vars(self):
             with self._monkeyImplicit():
                 self._callFUT()
 
-        self.assertNotEqual(_implicit_environ.DATASET_ID, GCD_DATASET_ID)
-        self.assertEqual(_implicit_environ.DATASET_ID, IMPLICIT_DATASET_ID)
+        self.assertNotEqual(_implicit_environ.get_default_dataset_id(),
```

Comment:
I put it in to "make sure" we didn't accidentally have `GCD_DATASET_ID == IMPLICIT_DATASET_ID`. Something like protecting against false positives by sending the wrong data.

You think the clear definition of the vars on 139-140 is sufficient?